pkg/sqlcache/store/store.go

```diff
@@ -524,6 +530,34 @@ func (s *Store) RegisterAfterDeleteAll(f func(txC transaction.Client) error) {
 	s.afterDeleteAll = append(s.afterDeleteAll, f)
 }
 
+func (s *Store) RegisterBeforeDropAll(f func(txC transaction.Client) error) {
+	s.beforeDropAll = append(s.beforeDropAll, f)
+}
+
+// DropAll effectively removes the store from the database. The store must be
+// recreated with NewStore.
+//
+// The store shouldn't be used once DropAll is called.
+func (s *Store) DropAll(ctx context.Context) error {
```

Comment:
This should really be called `DropBase`, and all the other variants of `dropall` should also move to `dropbase`.  "dropall" suggests code that would drop the tables `Foo`, `FooFields`, `FooLabels`, `FooIndices` and any other `Foo*` tables we create in the future.

Reply:
I'd be okay with just `Drop`, `DropBase` feels weird because we'd be calling `DropBase` but then more things would  be dropped in that call due to all registered functions.